src/mev_boost_tools.rs

```diff
@@ -0,0 +1,59 @@
+// ethereum specific
+use crate::mev_boost_tools::k256::schnorr::SigningKey;
+use ethers::prelude::{signer::SignerMiddleware, *};
+use ethers_flashbots::FlashbotsMiddleware;
+
+// general web utils
+use std::{sync::Arc, time::Duration};
+use url::Url;
+
+pub async fn initialize_mev_boost(
+    rpc_url: String,
+    tx_signer: String,
+    bundle_signer: String,
+    interval: Duration,
+) -> (
+    H160,
+    U256,
+    u64,
+    Arc<
+        SignerMiddleware<
+            FlashbotsMiddleware<Arc<Provider<Provider>>, Wallet<SigningKey>>,
+            Wallet<SigningKey>,
+        >,
+    >,
```

Comment:
You might be able to call this `impl Middleware` in the return type, but OK to leave as-is rn.